src/main/java/org/jitsi/videobridge/eventadmin/callstats/Activator.java

```diff
@@ -0,0 +1,171 @@
+     * starts/stops this implementation depending on the type of event i.e.
+     * REGISTERED/UNREGISTERING.
+     *
+     * @param ev the {@code ServiceEvent}
+     */
+    @Override
+    public void serviceChanged(ServiceEvent ev)
+    {
+        switch (ev.getType())
+        {
+        case ServiceEvent.REGISTERED:
+            if (bundleContext != null)
+            {
+                Object service;
+
+                try
+                {
+                    service
+                        = bundleContext.getService(ev.getServiceReference());
+                }
+                catch (IllegalArgumentException
+                        | IllegalStateException
+                        | SecurityException ex)
+                {
+                    service = null;
+                }
+                if (service instanceof CallStats)
+                {
+                    ConfigurationService cfg
+                        = ServiceUtils.getService(
+                                bundleContext,
+                                ConfigurationService.class);
+                    String bridgeId
+                        = ConfigUtils.getString(
+                                cfg,
+                                "io.callstats.sdk.CallStats.bridgeId",
+                                null);
+                    int interval
+                        = ConfigUtils.getInt(
+                                cfg,
+                                StatsManagerBundleActivator
+                                    .STATISTICS_INTERVAL_PNAME,
+                                StatsManagerBundleActivator
+                                    .DEFAULT_STAT_INTERVAL);
+
+                    // Update with per stats transport interval if available.
+                    interval
+                        = ConfigUtils.getInt(
+                                cfg,
+                                StatsManagerBundleActivator
+                                        .STATISTICS_INTERVAL_PNAME
+                                    + "."
+                                    + StatsManagerBundleActivator
+                                        .STAT_TRANSPORT_CALLSTATS_IO,
+                                interval);
+
+                    conferenceStatsHandler
+                        = new CallStatsConferenceStatsHandler();
+                    conferenceStatsHandler.start(
+                            (CallStats) service,
+                            bridgeId,
+                            interval);
+
+                    String[] topics = { "org/jitsi/*" };
+
+                    serviceRegistration
+                        = EventUtil.registerEventHandler(
+                                bundleContext,
+                                topics,
+                                conferenceStatsHandler);
+                }
+            }
+            break;
+
+        case ServiceEvent.UNREGISTERING:
```

Comment:
@damencho, shouldn't you be listening to UNREGISTERING of CallStats only here and not just any OSGi service? Like you do in REGISTERED?